bindings/python/scripts/parse.py

```diff
@@ -0,0 +1,279 @@
+        buildDir=compilation_database_path
+    )
+
+    # Get compiler arguments from the compilation database for the given file
+    compilation_commands = compilation_database.getCompileCommands(filename=filename)
+
+    """
+    - compilation_commands:
+        - An iterable object providing all the compilation commands available to build filename.
+        - type: <class 'clang.cindex.CompileCommands'>
+    - compilation_commands[0]:
+        - Since we have only one command per filename in the compile_commands.json, extract 0th element
+        - type: <class 'clang.cindex.CompileCommand'>
+    - compilation_commands[0].arguments:
+        - Get compiler arguments from the CompileCommand object
+        - type: <class 'generator'>
+    - list(compilation_commands[0].arguments)[1:-1]:
+        - Convert the generator object to list, and extract compiler arguments
+        - 0th element is the compiler name
+        - nth element is the filename
+    """
+
+    return list(compilation_commands[0].arguments)[1:-1]
+
+
+def parse_file(source, compilation_database_path=None):
+    """
+    Returns the parsed_info for a file
+
+    Parameters:
+        - source: Source to parse
+        - compilation_database_path: The path to `compile_commands.json`
+
+    Returns:
+        - parsed_info (dict)
+    """
+
+    # Create a new index to start parsing
+    index = clang.Index.create()
+
+    # Get compiler arguments
+    compilation_commands = get_compilation_commands(
+        compilation_database_path=compilation_database_path, filename=source,
+    )
+
+    """
+    - Parse the given source code file by running clang and generating the AST before loading
+    - option `PARSE_DETAILED_PROCESSING_RECORD`:
+        - Indicates that the parser should construct a detailed preprocessing record, 
+          including all macro definitions and instantiations.
+        - Required to get the `INCLUSION_DIRECTIVE`s.
+    """
+    source_ast = index.parse(
+        path=source,
+        args=compilation_commands,
+        options=clang.TranslationUnit.PARSE_DETAILED_PROCESSING_RECORD,
+    )
+
+    # Dictionary to hold a node's information
+    root_node = {
+        "cursor": source_ast.cursor,
+        "filename": source_ast.spelling,
+        "depth": 0,
+    }
+
+    # For testing purposes
+    # print_ast(root_node)
+
+    return generate_parsed_info(root_node)
+
+
+def main():
+    # Get command line arguments
+    args = utils.parse_arguments(script="parse")
+    for source in args.files:
+        source = utils.get_realpath(path=source)
+
+        # Parse the source file
+        parsed_info = parse_file(source, args.compilation_database_path)
+
+        # Output path for dumping the parsed info into a json file
+        output_filepath = utils.get_output_path(
+            source=source,
+            output_dir=utils.join_path(args.json_output_path, "json"),
+            split_from="pcl",
```

Comment:
`split_from` is another user input. This is technically base_path or something similar